### ISSUE TITLE
run: doesn't trigger obtaining container image on remote location during execution

### ISSUE THREAD

**Issue by yarikoptic** [ReproNim/reproman#442]
Container is a known to datalad-container extension container in a subdataset.  It is not present locally.  Running `niceman run` on a remote condor results in job crashing due to image file not being found. 

```shell
…nim/reproman-master/docs/usecases/bids-fmriprep-workflow-NP/out > declare | grep 'RM_'                                                                                                                      
RM_ORC=datalad-pair-run
RM_RESOURCE=smaug
RM_SUB=condor

…nim/reproman-master/docs/usecases/bids-fmriprep-workflow-NP/out > reproman run --follow -r "${RM_RESOURCE}" --sub "${RM_SUB}" --orc "${RM_ORC}" --jp container=containers/bids-mriqc data/bids data/mriqc participant,group
2019-07-24 15:16:23,618 [INFO   ] No root directory supplied for smaug; using '/home/yoh/.reproman/run-root' 
[INFO   ] Connecting ... 
ECDSA host key for IP address '129.170.233.9' not in list of known hosts.
[INFO   ] Considering to create a target dataset /home/yoh/proj/repronim/reproman-master/docs/usecases/bids-fmriprep-workflow-NP/out at /home/yoh/.reproman/run-root/1ab254ca-ad49-11e9-95c1-8019340ce7f2 of smaug 
[INFO   ] Fetching updates for <Dataset path=/home/yoh/proj/repronim/reproman-master/docs/usecases/bids-fmriprep-workflow-NP/out> 
.: smaug(+) [ssh://smaug/home/yoh/.reproman/run-root/1ab254ca-ad49-11e9-95c1-8019340ce7f2 (git)]                                                                                                                                  
[INFO   ] Adjusting remote git configuration 
[INFO   ] Considering to create a target dataset /home/yoh/proj/repronim/reproman-master/docs/usecases/bids-fmriprep-workflow-NP/out/containers at /home/yoh/.reproman/run-root/1ab254ca-ad49-11e9-95c1-8019340ce7f2/containers of smaug 
[INFO   ] Fetching updates for <Dataset path=/home/yoh/proj/repronim/reproman-master/docs/usecases/bids-fmriprep-workflow-NP/out/containers> 
.: smaug(+) [ssh://smaug/home/yoh/.reproman/run-root/1ab254ca-ad49-11e9-95c1-8019340ce7f2/containers (git)]                                                                                                                       
[INFO   ] Adjusting remote git configuration 
[INFO   ] Considering to create a target dataset /home/yoh/proj/repronim/reproman-master/docs/usecases/bids-fmriprep-workflow-NP/out/data/bids at /home/yoh/.reproman/run-root/1ab254ca-ad49-11e9-95c1-8019340ce7f2/data/bids of smaug 
[INFO   ] Fetching updates for <Dataset path=/home/yoh/proj/repronim/reproman-master/docs/usecases/bids-fmriprep-workflow-NP/out/data/bids> 
.: smaug(+) [ssh://smaug/home/yoh/.reproman/run-root/1ab254ca-ad49-11e9-95c1-8019340ce7f2/data/bids (git)]                                                                                                                        
[INFO   ] Adjusting remote git configuration 
[INFO   ] Considering to create a target dataset /home/yoh/proj/repronim/reproman-master/docs/usecases/bids-fmriprep-workflow-NP/out/data/mriqc at /home/yoh/.reproman/run-root/1ab254ca-ad49-11e9-95c1-8019340ce7f2/data/mriqc of smaug 
[INFO   ] Fetching updates for <Dataset path=/home/yoh/proj/repronim/reproman-master/docs/usecases/bids-fmriprep-workflow-NP/out/data/mriqc> 
.: smaug(+) [ssh://smaug/home/yoh/.reproman/run-root/1ab254ca-ad49-11e9-95c1-8019340ce7f2/data/mriqc (git)]                                                                                                                       
[INFO   ] Adjusting remote git configuration 
[INFO   ] Running post-update hooks in all created siblings 
[INFO   ] Publishing <Dataset path=/home/yoh/proj/repronim/reproman-master/docs/usecases/bids-fmriprep-workflow-NP/out/data/mriqc> to smaug 
[INFO   ] Publishing <Dataset path=/home/yoh/proj/repronim/reproman-master/docs/usecases/bids-fmriprep-workflow-NP/out/data/bids> to smaug                                                                                        
[INFO   ] Publishing <Dataset path=/home/yoh/proj/repronim/reproman-master/docs/usecases/bids-fmriprep-workflow-NP/out/containers> to smaug                                                                                       
[INFO   ] Publishing <Dataset path=/home/yoh/proj/repronim/reproman-master/docs/usecases/bids-fmriprep-workflow-NP/out> to smaug                                                                                                  
2019-07-24 15:16:42,546 [INFO   ] Making inputs available                                                                                                                                                                         
[INFO   ] Publishing <Dataset path=/home/yoh/proj/repronim/reproman-master/docs/usecases/bids-fmriprep-workflow-NP/out/containers> data to smaug 
2019-07-24 15:16:48,043 [INFO   ] Submitting 20190724-151622-831d 
2019-07-24 15:16:48,647 [INFO   ] Submitting /home/yoh/.reproman/run-root/1ab254ca-ad49-11e9-95c1-8019340ce7f2/.reproman/jobs/smaug/20190724-151622-831d/submit 
2019-07-24 15:16:49,157 [INFO   ] Job 20190724-151622-831d submitted as condor job 1824 
2019-07-24 15:16:49,246 [INFO   ] Registered job 20190724-151622-831d 
2019-07-24 15:16:49,325 [INFO   ] Waiting on job 1824: running 
2019-07-24 15:16:59,534 [INFO   ] Fetching results for 20190724-151622-831d 
2019-07-24 15:16:59,778 [INFO   ] Creating run commit in /home/yoh/proj/repronim/reproman-master/docs/usecases/bids-fmriprep-workflow-NP/out 
2019-07-24 15:17:00,973 [WARNING] 1 subjob failed. Check files in /home/yoh/proj/repronim/reproman-master/docs/usecases/bids-fmriprep-workflow-NP/out/.reproman/jobs/smaug/20190724-151622-831d/                                  
2019-07-24 15:17:00,973 [INFO   ] 20190724-151622-831d stderr: /home/yoh/proj/repronim/reproman-master/docs/usecases/bids-fmriprep-workflow-NP/out/.reproman/jobs/smaug/20190724-151622-831d/stderr.0 
2019-07-24 15:17:00,974 [INFO   ] Unregistered job 20190724-151622-831d 
reproman run --follow -r "${RM_RESOURCE}" --sub "${RM_SUB}" --orc "${RM_ORC}"  11.11s user 3.63s system 37% cpu 39.307 total

…nim/reproman-master/docs/usecases/bids-fmriprep-workflow-NP/out > cat /home/yoh/proj/repronim/reproman-master/docs/usecases/bids-fmriprep-workflow-NP/out/.reproman/jobs/smaug/20190724-151622-831d/stderr.0
ERROR  : Image path containers/images/bids/bids-mriqc--0.15.0.sing doesn't exist: No such file or directory
ABORT  : Retval = 255

…nim/reproman-master/docs/usecases/bids-fmriprep-workflow-NP/out > ls -Ll containers/images/bids/bids-mriqc--0.15.0.sing                                                                                     
ls: cannot access 'containers/images/bids/bids-mriqc--0.15.0.sing': No such file or directory

…nim/reproman-master/docs/usecases/bids-fmriprep-workflow-NP/out > ls -l containers/images/bids/bids-mriqc--0.15.0.sing 
lrwxrwxrwx 1 yoh yoh 142 Jul 23 08:55 containers/images/bids/bids-mriqc--0.15.0.sing -> ../../.git/annex/objects/MJ/8v/MD5E-s2928967711--b76d4d850bf94dbea13d0a3331451842.15.0/MD5E-s2928967711--b76d4d850bf94dbea13d0a3331451842.15.0

…nim/reproman-master/docs/usecases/bids-fmriprep-workflow-NP/out > ssh smaug ls -l /home/yoh/.reproman/run-root/1ab254ca-ad49-11e9-95c1-8019340ce7f2/containers/images/bids/bids-mriqc--0.15.0.sing
ECDSA host key for IP address '129.170.233.9' not in list of known hosts.
lrwxrwxrwx 1 yoh yoh 142 Jul 24 15:17 /home/yoh/.reproman/run-root/1ab254ca-ad49-11e9-95c1-8019340ce7f2/containers/images/bids/bids-mriqc--0.15.0.sing -> ../../.git/annex/objects/MJ/8v/MD5E-s2928967711--b76d4d850bf94dbea13d0a3331451842.15.0/MD5E-s2928967711--b76d4d850bf94dbea13d0a3331451842.15.0

…nim/reproman-master/docs/usecases/bids-fmriprep-workflow-NP/out > ssh smaug ls -Ll /home/yoh/.reproman/run-root/1ab254ca-ad49-11e9-95c1-8019340ce7f2/containers/images/bids/bids-mriqc--0.15.0.sing 
ECDSA host key for IP address '129.170.233.9' not in list of known hosts.
ls: cannot access '/home/yoh/.reproman/run-root/1ab254ca-ad49-11e9-95c1-8019340ce7f2/containers/images/bids/bids-mriqc--0.15.0.sing': No such file or directory
```

niceman v0.1.0-539-g2c87dcc from #408

**Comment by yarikoptic** [comment]
```
…riprep-workflow-NP/out > cat stdout.0 
[ReproMan] pre-command...
[ReproMan] executing command containers/scripts/singularity_cmd run containers/images/bids/bids-mriqc--0.15.0.sing data/bids data/mriqc participant,group
[ReproMan] ... within /home/yoh/.reproman/run-root/1ab254ca-ad49-11e9-95c1-8019340ce7f2
[ReproMan] post-command...
```
so it seems that `datalad container-run` is not used, and image is not retrieved first

**Comment by yarikoptic** [comment]
I don't think this issue is still not addressed -- haven't had such one for a while

**Comment by kyleam** [comment]
Yes, I believe this was addressed by 002842208 (BF: orchestrators: Trigger git-annex autoenabling, 2019-08-05), merged with 0b5bc6163 (Merge pull request #408 from kyleam/run-subjobs, 2019-08-09).
